crates/trade/src/cfd.rs

```diff
@@ -54,3 +58,61 @@ pub fn calculate_short_liquidation_price(leverage: Decimal, price: Decimal) -> D
 
     price * leverage / (leverage - Decimal::ONE)
 }
+
+// TODO: This was copied from ItchySats and adapted; we need tests for this!
+/// Compute the payout for the given CFD parameters at a particular `closing_price`.
+///
+/// The PNL, both as a `bitcoin::SignedAmount` and as a percentage, is also returned for
+/// convenience.
+///
+/// The `Position` is determined based on the `FeeAccount`.
+///
+/// These formulas are independent of the inverse payout curve implementation and are therefore
+/// theoretical. There could be slight differences between what we return here and what the payout
+/// curve determines.
+pub fn calculate_payout_at_price(
+    opening_price: f64,
+    closing_price: f64,
+    quantity: f64,
+    long_leverage: f64,
+    short_leverage: f64,
+    direction: Direction,
+) -> Result<u64> {
+    let opening_price = Decimal::try_from(opening_price).expect("price to fit into decimal");
```

Comment:
🤔 Shouldn't we use the same payout function here as we use when closing the position?

Reply:
Good point: Either this function calculates the `PnL at price` or the `payout at price`. For the latter one we would need to use the payout funciton to determine the correct payout. 
I'd go for the former one

Reply:
❓ hmm.. is that the way it's typically done? As a user I would like to see how much money I would get if I sell / buy right now. What's the difference between `PnL at price` and `payout at price`

side note: I know it can still not be guaranteed as there has to be an offer for the market order, but still that would be the closest approximation.

Reply:
`PnL` is the profit and loss, so it's on top of your collateral. 
I think it depends on the trader if they would like to see `PnL` or payout - but we can easily calculate the payout as `collateral + pnl` - if we assume that fees are already calculated in. 

I would display the pnl for no and see what users say

Reply:
but still the underlying calculation is different. the way you describe it is that `payout at price` is actually part of the `pnl at price`. `PnL at price` = collateral + `payout at price`